src/components/InventoryPage/DonationModal.jsx

```diff
@@ -0,0 +1,405 @@
+import React, { useState, useEffect } from 'react';
+import { PropTypes } from 'prop-types';
+
+import {
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Button,
+  Box,
+  Flex,
+  Text,
+  Input,
+  Stack,
+  InputLeftAddon,
+  InputRightAddon,
+  InputGroup,
+  Textarea,
+  useDisclosure,
+  Tr,
+  Modal,
+  Td,
+} from '@chakra-ui/react';
+
+import { PNPBackend } from '../../utils/utils';
+import { getPictureFromDB } from '../../utils/InventoryUntils';
+import './InventoryPage.module.css';
+// import updateDonationStatus from '../../utils/InventoryUntils';
+
+const DonationModal = prop => {
+  const { props } = prop;
+  const {
+    id,
+    addressStreet,
+    addressUnit,
+    addressCity,
+    addressZip,
+    firstName,
+    lastName,
+    email,
+    phoneNum,
+    notes,
+    submittedDate,
+    status,
+  } = props;
+
+  const [image, setImage] = useState([]);
+
+  useEffect(async () => {
+    const result = await getPictureFromDB(id);
+    setImage(result[0]);
+    // console.log(result[0]);
+  }, []);
+
+  // console.log(props);
+  // console.log(id);
+
+  const [canEditInfo, setCanEditInfo] = useState(false);
+  const [first, setFirstName] = useState(firstName);
+  // const [last, setLastName] = useState(lastName);
+  const [newemail, setEmail] = useState(email);
+  const [phone, setPhoneNum] = useState(phoneNum);
+  const [street, setStreet] = useState(addressStreet);
+  const [unit, setUnit] = useState(addressUnit);
+  const [city, setCity] = useState(addressCity);
+  const [zip, setZip] = useState(addressZip);
+  const [newnote, setNotes] = useState(notes);
```

Comment:
Instead of having a `useState` for each variable, let's use [react hook form](https://react-hook-form.com/get-started) to register the variables.

Reply:
There should also be validation for the email (proper formatting) and zipcode (exactly 5 digits), and this can be done using the [yup](https://github.com/jquense/yup) library.

Reply:
Refer to the [donation form](https://github.com/ctc-uci/patriots-and-paws-frontend/blob/dev/src/components/DonationForm/DonationForm.jsx) for how to set up the variables and add validation